service/storage_service.cc

```diff
@@ -392,7 +387,12 @@ void storage_service::prepare_to_join(std::vector<inet_address> loaded_endpoints
         ss._local_host_id = local_host_id;
     }).get();
     auto features = get_config_supported_features();
-    _token_metadata.update_host_id(local_host_id, get_broadcast_address());
+    if (!replacing_a_node_with_diff_ip) {
+        // Replacing node with a different ip should own the host_id only after
+        // the replacing node becomes NORMAL status. It is updated in
+        // handle_state_normal().
```

Comment:
We set the host_id in gossip a few lines below. We just can't set it in _token_metadata yet because the host_id is still owned by the old dead node as the result of the cluster sees.

Reply:
So token_metadata only has to know about host ids of tokens which form the old ring? We don't have to have a host if for pending or replacing tokens?

Reply:
The pending and replacing tokens are the tokens of the node being replaced. The host id to ip mapping is still being host id <-> old_ip during the replace operation.